integration-tests/tests/bombastic.rs

```diff
@@ -181,3 +182,83 @@ async fn test_invalid_encoding(context: &mut BombasticContext) {
     assert_eq!(response.status(), StatusCode::BAD_REQUEST);
     assert_eq!(response.headers().get("accept-encoding").unwrap(), &"bzip2, zstd");
 }
+
+#[test_context(BombasticContext)]
+#[tokio::test]
+#[ntest::timeout(60_000)]
+async fn test_upload_sbom_existing_without_change(context: &mut BombasticContext) {
+    let input = serde_json::from_str(include_str!("../../bombastic/testdata/my-sbom.json")).unwrap();
+    let id = "test-upload";
+    let api_end_point = format!("api/v1/sbom?id={id}");
+    upload_sbom(context.port, id, &input, &context.provider).await;
+    let response: Value = get_response(context.port, &api_end_point, StatusCode::OK, &context.provider)
+        .await
+        .into();
+    assert_eq!(input, response, "Content mismatch between request and response");
+    upload_sbom(context.port, id, &input, &context.provider).await;
+    let response: Value = get_response(context.port, &api_end_point, StatusCode::OK, &context.provider)
+        .await
+        .into();
+    assert_eq!(
+        input, response,
+        "Content mismatch between request and response after update"
+    );
+}
+
+#[test_context(BombasticContext)]
+#[tokio::test]
+#[ntest::timeout(60_000)]
+async fn test_upload_sbom_existing_with_change(context: &mut BombasticContext) {
+    let mut input1 = serde_json::from_str(include_str!("../../bombastic/testdata/my-sbom.json")).unwrap();
+    let id = "test-upload-with-change";
+    let api_end_point = format!("api/v1/sbom?id={id}");
+    upload_sbom(context.port, id, &input1, &context.provider).await;
+    let response1: Value = get_response(context.port, &api_end_point, StatusCode::OK, &context.provider)
+        .await
+        .into();
+    assert_eq!(input1, response1, "Content mismatch between request and response");
+    input1["metadata"]["component"]["name"] = Value::String(String::from("update-sbom-name"));
+    upload_sbom(context.port, id, &input1, &context.provider).await;
+    let response2: Value = get_response(context.port, &api_end_point, StatusCode::OK, &context.provider)
+        .await
+        .into();
+    assert_eq!(
+        input1, response2,
+        "Content mismatch between request and response after update"
+    );
+}
+
+#[test_context(BombasticContext)]
+#[tokio::test]
+#[ntest::timeout(60_000)]
+async fn test_upload_empty_json(context: &mut BombasticContext) {
+    //Known issue - Respose 200 instead of 400
+    let input: serde_json::Value = serde_json::json!({});
+    let id = "empty-json-upload";
+    let api_end_point = format!("api/v1/sbom?id={id}");
+    upload_sbom(context.port, id, &input, &context.provider).await;
+    get_response(context.port, &api_end_point, StatusCode::BAD_REQUEST, &context.provider).await;
```

Comment:
This is a tricky one, because we don't validate the content of the SBOM on the upload. To avoid the possibility of a DoS attack, we avoid loading the entire SBOM into memory. Instead, the indexer does asynchronous to the upload.

I guess we have to decide if this test is worth the additional code complexity to "peek" at the payload to see if it's small enough to validate. Any thoughts @lulf?

Reply:
I think the not loading the entire SBOM into memory is more important right now. Though I think maybe the test can verify that there is going to be an event in the sbom-failed topic with the appropriate error that the document could not be indexed?